ma_gym/envs/traffic_junction/traffic_junction.py

```diff
@@ -44,20 +47,47 @@ class TrafficJunction(gym.Env):
     """
     metadata = {'render.modes': ['human', 'rgb_array']}
 
-    def __init__(self, grid_shape=(14, 14), n_agents=10, n_max=10, arrive_prob=0.5, full_observable=False):
+    def __init__(self, grid_shape=(14, 14), step_cost=0, n_max=4, rcoll=-10, arrive_prob=0.5, full_observable=False):
         self._grid_shape = grid_shape
-        self.n_agents = n_agents
+        self.n_agents = n_max
```

Comment:
n_agents will be a constant in the case rather than being "n_max" since we haven't defined rest for the code for more than 4 agents.

So, let's keep `self.n_agents=4`. Also, make it private and accessible through a property decorator so that it cannot be changed from outside.